Adaptors/MongoDB/src/ResultTable.cs

```diff
@@ -94,30 +94,29 @@ public class ResultTable : IResultTable
     }
   }
 
-  /// <inheritdoc />
-  public async Task<Core.Common.Storage.Result> GetResult(string            sessionId,
-                                                          string            key,
-                                                          CancellationToken cancellationToken = default)
+  public async IAsyncEnumerable<Core.Common.Storage.Result> GetResults(string                                     sessionId,
+                                                                       IEnumerable<string>                        keys,
+                                                                       [EnumeratorCancellation] CancellationToken cancellationToken = default)
   {
-    using var activity = activitySource_.StartActivity($"{nameof(GetResult)}");
-    activity?.SetTag($"{nameof(GetResult)}_sessionId",
+    using var activity = activitySource_.StartActivity($"{nameof(GetResults)}");
+    activity?.SetTag($"{nameof(GetResults)}_sessionId",
                      sessionId);
-    activity?.SetTag($"{nameof(GetResult)}_key",
-                     key);
     var sessionHandle    = sessionProvider_.Get();
     var resultCollection = resultCollectionProvider_.Get();
 
-    try
-    {
-      return await resultCollection.AsQueryable(sessionHandle)
-                                   .Where(model => model.Id == Result.GenerateId(sessionId,
-                                                                                 key))
-                                   .SingleAsync(cancellationToken)
-                                   .ConfigureAwait(false);
-    }
-    catch (InvalidOperationException)
+    var cursor = await resultCollection.AsQueryable(sessionHandle)
+                                       .Where(model => keys.Contains(model.Name) && model.SessionId == sessionId)
+                                       .ToCursorAsync(cancellationToken)
+                                       .ConfigureAwait(false);
+
+    while (await cursor.MoveNextAsync(cancellationToken)
```

Comment:
Could this be an extension method of a cursor? I expect this to be used many times.